src/systems/python_system_loader/PythonSystemLoader.cc

```diff
@@ -0,0 +1,198 @@
+  py::module_ sysModule;
+  try
+  {
+    py::module::import(gzSimModule.c_str());
+    py::module::import(sdformatModule.c_str());
+    sysModule = py::module_::import("sys");
+  }
+  catch (const pybind11::error_already_set &_err)
+  {
+    gzerr << "Error while loading required modules:\n"
+          << _err.what() << std::endl;
+    return;
+  }
+
+  // Add GZ_SIM_SYSTEM_PLUGIN_PATH and other default plugin 
+  // locations to PYTHONPATH
+  try
+  {
+    SystemLoader systemLoader;
+    for (const auto &pluginPath : systemLoader.PluginPaths())
+    {
+      sysModule.attr("path").attr("append")(pluginPath);
+    }
+  }
+  catch (const std::runtime_error &_err)
+  {
+    gzerr << _err.what() << std::endl;
+    return;
+  }
+
+  try
+  {
+    this->pythonModule = py::module_::import(moduleName.c_str());
+  }
+  catch (const pybind11::error_already_set &_err)
+  {
+    gzerr << "Error while loading module " << moduleName << "\n"
+          << _err.what() << "\n";
+
+    gzerr << "Searched in:\n";
+    for (const auto &path : sysModule.attr("path"))
+    {
+      gzerr << "  \"" << path << "\"\n";
+    }
+    return;
+  }
+
+  try
+  {
+    py::object getSystem = this->pythonModule.attr("get_system");
+    if (!getSystem)
+    {
+      gzerr << "Could not call 'get_system' on the module" << moduleName << "\n";
+      return;
+    }
+    this->pythonSystem = getSystem();
+  }
+  catch (const pybind11::error_already_set &_err)
+  {
+    gzerr << _err.what() << std::endl;
+    return;
+  }
+  if (py::hasattr(this->pythonSystem, "configure"))
+  {
+    try
+    {
+      // _ecm and _eventMgr are passed in as pointers so pybind11 would use the
+      // write `return_value_policy`. Otherwise, it would use
+      // `return_value_policy = copy`.
+      this->pythonSystem.attr("configure")(_entity, _sdf, &_ecm, &_eventMgr);
+    }
+    catch (const pybind11::error_already_set &_err)
+    {
+      gzerr << _err.what() << std::endl;
+      return;
+    }
+  }
+  // TODO(azeey) Add support for ConfigureParameters
+  if (py::hasattr(this->pythonSystem, "pre_update"))
+  {
+    this->preUpdateMethod = this->pythonSystem.attr("pre_update");
+  }
+  if (py::hasattr(this->pythonSystem, "update"))
+  {
+    this->updateMethod = this->pythonSystem.attr("update");
+  }
+  if (py::hasattr(this->pythonSystem, "post_update"))
+  {
+    this->postUpdateMethod = this->pythonSystem.attr("post_update");
+  }
+  if (py::hasattr(this->pythonSystem, "reset"))
+  {
+    this->resetMethod = this->pythonSystem.attr("reset");
+  }
+  this->validConfig = true;
+}
+
+//////////////////////////////////////////////////
+template <typename ...Args>
+void PythonSystemLoader::CallPythonMethod(py::object _method, Args&&... _args)
+{
+  if (!this->validConfig)
+  {
+    return;
+  }
+
+  if (_method)
+  {
+    try
+    {
+      _method(std::forward<Args>(_args)...);
+    }
+    catch (const pybind11::error_already_set &_err)
+    {
+      gzerr << _err.what() << std::endl;
```

Comment:
Will the error message contain the name of the method and python module here?  It would certainly help with tracking down errors, otherwise they will all have the same name, correct?

Reply:
It should, but I'll double check.

Reply:
I threw an exception in the python file and this is what I got
```
[Err] [PythonSystemLoader.cc:174] NameError: name 'entity' is not defined

At:
  ~/ws/harmonic/src/gz-sim/python/test/plugins/test_model_system.py(51): pre_update

```
I think that's sufficient.